assets/js/color.js

```diff
@@ -0,0 +1,132 @@
+/**
+ * Color Calculations.
+ *
+ * @since 1.0.0
+ *
+ * @param {string} backgroundColor - The background color.
+ * @param {Number} accentHue - The hue for our accent color.
+ *
+ * @return {Object} - this
+ */
+function _twentyTwentyColor( backgroundColor, accentHue ) {
+
+	// Set the object properties.
+	this.backgroundColor = backgroundColor;
+	this.accentHue       = accentHue;
+	this.bgColorObj      = new Color( backgroundColor );
+	this.textColorObj    = this.bgColorObj.getMaxContrastColor();
+	this.textColor       = this.textColorObj.toCSS();
+	this.isDark          = this.bgColorObj.toLuminosity() < 0.5;
+	this.isLight         = ! this.isDark;
+
+	// Return the object.
+	return this;
+}
+
+/**
+ * Builds an array of Color objects based on the accent hue.
+ * For improved performance we only build half the array
+ * depending on dark/light background-color.
+ *
+ * @since 1.0.0
+ *
+ * @return {Object} - this
+ */
+_twentyTwentyColor.prototype.setAccentColorsArray = function() {
+	var minSaturation    = 40,
+		maxSaturation    = 100,
+		minLightness     = this.isDark ? 40 : 0,
+		maxLighness      = this.isDark ? 100 : 60,
+		stepSaturation   = 4,
+		stepLightness    = 4,
```

Comment:
We can tweak these variables here to change the accuracy of the script and how it calculates things.

Reply:
The input color doesn't have a saturation, it's hue only.

Reply:
The implementation here was improved a lot since this comment, I think we can close this one :)